### ISSUE TITLE
Improper TEXT decoding for newlines?

### ISSUE THREAD

**Issue by devinrsmith** [OpenHFT/Chronicle-Wire#37]
The following code outputs

```
obj1.equals(obj2): true
output: [hello\nworld
]
obj2.equals(obj3): false
```

I expect that `obj2.equals(obj3)` should be `true`.

```java
import com.google.common.base.Preconditions;
import net.openhft.chronicle.bytes.Bytes;
import net.openhft.chronicle.core.io.IORuntimeException;
import net.openhft.chronicle.wire.Marshallable;
import net.openhft.chronicle.wire.WireIn;
import net.openhft.chronicle.wire.WireOut;
import net.openhft.chronicle.wire.WireType;

import java.nio.ByteBuffer;

/**
 * Created by dsmith on 11/11/16.
 */
public class MarshallableObj implements Marshallable {
    private final StringBuilder builder = new StringBuilder();

    public void clear() {
        builder.setLength(0);
    }

    public void append(CharSequence cs) {
        builder.append(cs);
    }

    @Override
    public void readMarshallable(WireIn wire) throws IORuntimeException {
        builder.setLength(0);
        Preconditions.checkState(wire.getValueIn().textTo(builder) != null);
    }

    @Override
    public void writeMarshallable(WireOut wire) {
        wire.getValueOut().text(builder);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MarshallableObj that = (MarshallableObj) o;

        return builder.toString().equals(that.builder.toString());
    }

    @Override
    public int hashCode() {
        return builder.toString().hashCode();
    }

    @Override
    public String toString() {
        return builder.toString();
    }

    public static void main(String[] args) {
        final WireType wireType = WireType.TEXT;
        final String exampleString = "hello\nworld";

        final MarshallableObj obj1 = new MarshallableObj();
        final MarshallableObj obj2 = new MarshallableObj();
        final MarshallableObj obj3 = new MarshallableObj();

        obj1.append(exampleString);
        obj2.append(exampleString);

        System.out.println("obj1.equals(obj2): " + obj1.equals(obj2));

        final Bytes<ByteBuffer> bytes = Bytes.elasticByteBuffer();
        obj2.writeMarshallable(wireType.apply(bytes));

        final String output = bytes.toString();
        System.out.println("output: [" + output + "]");

        obj3.readMarshallable(wireType.apply(Bytes.from(output)));

        System.out.println("obj2.equals(obj3): " + obj2.equals(obj3));
    }
}
```

Strangely enough, if you use `"hello\\ncruel\nworld"` or `""hello\ncruel\\nworld"` it outputs `true`. Also, `WireType.BINARY` works in all cases. I'm assuming that I'm using the interface correctly since it works w/ `BINARY`.

**Comment by peter-lawrey** [comment]
Most likely this is due to the \n being escaped incorrectly. Thank you for
the test. I will add it.

**Comment by devinrsmith** [comment]
Hoping this is only a bug in decoding and not encoding. Otherwise I have some migrations to do!

**Comment by peter-lawrey** [comment]
I have converted the test in a junit test WireBug37Test, can you review this?

**Comment by devinrsmith** [comment]
LGTM.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.22ea11](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.22ea11), [BOM-2.22ea69](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea69)
